src/sentry/identity/discord/provider.py

```diff
@@ -0,0 +1,9 @@
+from sentry.auth.providers.oauth2 import OAuth2Provider
+
+
+class DiscordIdentityProvider(OAuth2Provider):
+    key = "discord"
+    name = "Discord"
+
+    # We don't actually use OAuth2 for Discord identities, so this class is
```

Comment:
Does it need to extend `OAuth2Provider` then? Can it just extend `Provider`?

Reply:
Yes it can! Thanks for the suggestion @evanpurkhiser 😄

Reply:
Why doesn't it use OAuth2 btw? What procedure does it use for identity linking?

Reply:
My thinking is that we only need their Discord user id. We tie that to Sentry user via the Identity and when that Discord user takes actions on a notification in Discord, we'll know who's doing it. We don't need to go through the OAuth2 flow to get this user ID, Discord gives it to us when they use the `/link` command. 

Lmk if my thinking is flawed or there's some other reason to do the whole flow.

Reply:
I don't know enough about how this integration works to be able to advise on that I think.

Would be happy to sit down and walk through it all some time if you'd like though!